cmd/kube-proxy/app/server.go

```diff
@@ -335,7 +348,33 @@ with the apiserver API to configure the proxy.`,
 			verflag.PrintAndExitIfRequested()
 			cmdutil.CheckErr(opts.Complete())
 			cmdutil.CheckErr(opts.Validate(args))
-			cmdutil.CheckErr(opts.Run())
+
+			entCh := make(chan interface{})
+			go func() {
+				for {
+					select {
+					case e := <-opts.event:
+						switch e.Op {
+						case fsnotify.Write:
+							entCh <- true
+						}
+					}
+				}
+			}()
+
+			cmdutil.CheckErr(rerun.ReRun(
```

Comment:
This seems very delicate to me.  If other goroutines are spawned, deep inside this, who is going to clean those up?  Who is going to release allocated memory?  I think you really just want to catch the event, exit cleanly, and let the kubelet restart you

Reply:
agree, this change (and the ReRun change) makes the scheduler execute flow almost incomprehensible. If the top-level kube-proxy command optionally shuts down the proxy and cleanly exits on config file change, that seems sufficient.

I expected more of the following:
* opts adds a `Stop()` function that shuts down scheduler processes cleanly (might require plumbing stop channels in `Run()`)
* opts contains an optional `OnConfigChange func()` field
* if present, it sets up the file listener and calls `OnConfigChange()` when the file changes
* top-level kube-proxy command (optionally) sets opts.OnConfigChange to `func() { opts.Stop(); os.Exit(0); }`, etc

That keeps the top-level flow very straightforward, avoids deeply embedding os.Exit() calls, allows other consumers of kubeproxy opts to hook in different behavior on config file change, etc.